src/camera_capture/camera_capture/camera_capture.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""ROS2 CSI Camera Image Publisher.
+This script publishes csi camera image to a ROS2 topic in sensor_msgs.msg/Image format.
+Example:
+        $ colcon build --symlink-install
+        $ ros2 launch camera_snap_shot camera_snap_shot.launch.py
+"""
+
+# ___Import Modules:
+import os
+import cv2
+import json
+import numpy as np
+
+import rclpy
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+from ament_index_python.packages import get_package_share_directory
+from std_msgs.msg import Bool
+from cv_bridge import CvBridge
+
+from . import PACKAGE_NAME
+
+
+# ___Global Variables:
+SETTINGS = os.path.join(
+    get_package_share_directory("camera_capture"), "config/camera_capture_settings.json"
+)
+with open(SETTINGS) as fp:
+    json_settings = json.load(fp)
+
+
+# __Functions:
+def gstreamer_pipeline(
+    capture_width: str = str(json_settings["capture_width_livestream"]),
+    capture_height: str = str(json_settings["capture_height_livestream"]),
+    framerate: str = str(json_settings["framerate"]),
+):
+    return (
+        "nvarguscamerasrc ! "
+        "video/x-raw(memory:NVMM), "
+        f"width=(int){capture_width}, height=(int){capture_height}, "
+        f"format=(string)NV12, framerate=(fraction){framerate}/1 ! "
+        "nvvidconv ! "
+        "video/x-raw, format=(string)BGRx ! "
+        "videoconvert ! "
+        "video/x-raw, format=(string)BGR ! "
+        "appsink"
+    )
+
+
+# __Classes:
+class CameraPublisher(Node):
+    """this class captures immages from a CSI camera and publishes it as a livestream or snapshots to a topic"""
+
+    def __init__(
+        self,
+        publish_livestream_topic: str,
+        publish_snapshot_topic: str,
+        snapshot_trigger_topic: str,
+        livestream_state_topic: str,
+    ):
+        super().__init__(PACKAGE_NAME)
+
+        # initialize publisher & subscirbers
+        self.pub_cam_snapshot = self.create_publisher(Image, publish_snapshot_topic, 1)
+        self.pub_cam_livestream = self.create_publisher(
+            Image, publish_livestream_topic, 1
+        )
+        self.sub_cam_livestream_state = self.create_subscription(
+            Bool, livestream_state_topic, self.start_livestream_callback, 1
+        )
+        self.sub_cam_snapshot_trigger = self.create_subscription(
+            Bool, snapshot_trigger_topic, self.capture_snapshot_callback, 1
+        )
+
+        self.cap = cv2.VideoCapture(gstreamer_pipeline())
+        self.bridge = CvBridge()
+        self.image_location = f"{json_settings['image_location']}"
+        self.image_counter = 0
+
+    def start_livestream_callback(self, topic_msg: Bool):
```

Comment:
The function name does not really reflect its function, as it starts and stops the livestream. Would `livestream_state_callback` be an option for you?